src/ytdl_sub/ytdl_additions/enhanced_download_archive.py

```diff
@@ -674,6 +678,22 @@ def save_file_to_output_directory(
             copy_file=copy_file,
         )
 
+        # Set mtime if preserve_mtime is enabled and we have an entry with upload_date
+        if preserve_mtime and entry and not self._file_handler.dry_run:
+            upload_date = entry.get(v.upload_date, str)
```

Comment:
Let's actually use `ytdl_sub_keep_files_date_eval` here.

This is a configurable variable that lets people use either upload date or release date. Some sites only support one but not the other. This should give mtime support to any of them.

The format is standardized, i.e. YYYY-mm-dd. Will need to update below as well.